### PR TITLE
SAK-42019 Convert Samigo settings pages to tab layout

### PR DESCRIPTION
**Pull request by davidpbauer**
This change converts the jquery ui accordion used for Samigo settings to
the jquery ui tabs. The layout is more condusive to working through all
of the settings and provides quicker access and better awareness while
and instructor sets up their assessment. All settings have remained in the
same bucket and the tab names are taken from the accordion titles. There
is a new section "Publish" that contains some information and the save
settings and publish button. This is in a new tab so that users don't accidentally
publish an exam before they are ready. The save and cancel buttons remain at the
bottom of the page.

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java [hunk 1]
```diff
@@ -203,6 +203,8 @@
   private ExtendedTime extendedTime;
   private ExtendedTime transitoryExtendedTime;
   private boolean editingExtendedTime = false;
+
+  private int activeTabIndex = 0;
```
Comment (bjones86) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java]: Can we make these values named constants? It'll be easier to understand the code in the future when this PR isn't so fresh in our minds. Magic numbers are a little ambiguous when reading out of context: "What does `0` represent? What does `2` represent?", etc.

Also if you put a Lombok `@Getter` and `@Setter` on this, then you don't have to manually write the getter and setter below.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/PublishedAssessmentSettingsBean.java [hunk 2]
```diff
@@ -202,6 +202,8 @@
   private ExtendedTime extendedTime;
   private ExtendedTime transitoryExtendedTime;
   private boolean editingExtendedTime = false;
+
+  private int activeTabIndex = 0;
```
Comment (bjones86) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/PublishedAssessmentSettingsBean.java]: Same things here.

**Comment by davidpbauer** [comment]
Thanks for the feedback. I've amended my commit to address the suggestion you made about the active tab index. Let me know if I've interpreted your comment correctly or if there is anything else you'd like me to change.

**Comment by ottenhoff** [comment]
I added comments to the JIRA. I'd like to hear more opinions before proceeding.

**Comment by davidpbauer** [comment]
@bjones86 I've removed the icons on the tabs and the published tab. The publish button was reverted back to the bottom of the page next to the other buttons.

**Comment by adrianfish** [comment]
Needs some usability testing before any kind of decision can be made.

**Pull request closed without merging**